tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs

```diff
@@ -32,9 +33,46 @@ public static List<OperationsViewModel> FillDotButtonOperations(ComputeSystem co
             operations.Add(new OperationsViewModel("Delete", "\uE74D", computeSystem.DeleteAsync));
         }
 
-        // ToDo: Correct the function used
-        // operations.Add(new OperationsViewModel("Pin To Taskbar", "\uE718", computeSystem.));
-        // operations.Add(new OperationsViewModel("Add to Start Menu", "\uE8A9", computeSystem.));
+        return operations;
+    }
+
+    public static async Task<List<OperationsViewModel>> FillDotButtonPinOperationsAsync(ComputeSystem computeSystem)
+    {
+        var supportedOperations = computeSystem.SupportedOperations;
+        var operations = new List<OperationsViewModel>();
+        if (supportedOperations.HasFlag(ComputeSystemOperations.PinToTaskbar))
+        {
+            var pinResultTaskbar = await computeSystem.GetIsPinnedToTaskbarAsync();
+            if (pinResultTaskbar.Result.Status == ProviderOperationStatus.Success)
+            {
+                if (pinResultTaskbar.IsPinned)
+                {
+                    operations.Add(new OperationsViewModel("Unpin From Taskbar", "\uE77A", computeSystem.UnpinFromTaskbarAsync));

```

Comment:
Strings must be localized (lines 50, 54, 66, 70).

Reply:
Done.